### PR TITLE
Updated documentation for k-Means and PolyReg

### PR DESCRIPTION
**Pull request by ajdapretnar**
@PrimozGodec Please check if you agree with the changes. I've tried to make the documentation consistent with Orange3, but if you prefer some lines from the original, I can change it back.

### CODE REVIEW

File: doc/widgets/kmeans.rst [hunk 1]
```diff
@@ -1,86 +1,105 @@
...
 ~~~~~~
-- **Data**
+
+-  **Data**
+
 Input data set.
 
-Outputs
-~~~~~~~
+**Outputs**:
```
Comment (PrimozGodec) [inline doc/widgets/kmeans.rst]: @ajdapretnar isn't here correct 

    Outputs
    ~~~~~~

File: doc/widgets/kmeans.rst [hunk 2]
```diff
@@ -1,86 +1,105 @@
...
-**Reasign membership** button.
+Then we repeat these two steps until the algorithm converges. This is the final result.
 
-.. figure:: images/kmeansstep3.png
+.. figure:: images/interactive-kmeans-step4.png
 
-Then we repeat that two steps until algorithm converges. It is final result.
+Perhaps we are not satisfied with the result because we noticed that maybe classification into 4 clusters would be better.
+So we decided to add a new centroid. We can do this by increasing the number of centroids in the control menu or with a click on
+the position in the graph where we want to place the centroid. We decided to add it with a click. The new centroid is the red one.
```
Comment (PrimozGodec) [inline doc/widgets/kmeans.rst]: This is my mistake from before. The last sentence would be: 

    The new centroid is the orange one.

File: doc/widgets/kmeans.rst [hunk 3]
```diff
@@ -1,86 +1,105 @@
...
 
-.. figure:: images/kmeanssteplast.png
+.. figure:: images/interactive-kmeans-step5.png
 
-But we are not satisfied with result because we noticed that maybe classification in 4 clusters would be better.
-So we decided to add a new centroid. We can make it with increasing number of centroids in left menu or with click on
-position in graph where we want to place the centroid. We decided to add it with click. New centroid is red one.
+Now we can repeat running the algorithm until it converges again, but before that we will move the green centroid to
```
Comment (PrimozGodec) [inline doc/widgets/kmeans.rst]: I think orange centroid were moved.

File: doc/widgets/univariatepolynomialregression.rst [hunk 4]
```diff
@@ -40,40 +41,45 @@ Regressor coefficients if it has them.
...
+In the widget, `polynomial expansion <https://en.wikipedia.org/wiki/Polynomial_expansion>`__ can be set. 
+Polynomial expansion is a regulation of the degree of polynom that is used to transform the input data and has an effect 
+on the shape of a curve. If polynomial expansion is set to 1 it means that untransformed data are used in the regression.
 
-.. figure:: images/polyregressionmain.png
+.. figure:: images/polynomial-regression-stamped.png
 
-1. Provide data with at least two continuous attributes on input. Also provide any regression learner on input (if
-   not provided *Linear Regression* is used.
+1. Regressor name. Input data needs least two continuous attributes.
```
Comment (PrimozGodec) [inline doc/widgets/univariatepolynomialregression.rst]: Maybe we can move `Input data needs least two continuous attributes.` `data` part in `signals`

File: doc/widgets/univariatepolynomialregression.rst [hunk 5]
```diff
@@ -40,40 +41,45 @@ Regressor coefficients if it has them.
...
 Example
 -------
 
-With *File* widget we loaded *Iris* data set. We also connect *Linear Regression* learner as show on figure above.
-In widget we selected *petal length* as an **Input** feature and *petal width* as a **Target**.
-We set **Polynomial expansion** to 1 what gives us linear regression line. Result is show on the figure below.
+.. figure:: images/polyregressionmain.png
+
+We loaded *iris* data set with the :doc:`File<../orange3/doc/visual-programming/source/widgets/data/file>` widget. 
+Then we connected :doc:`Linear Regression<../orange3/doc/visual-programming/source/widgets/regression/linear>` learner to the **File** widget.
```
Comment (PrimozGodec) [inline doc/widgets/univariatepolynomialregression.rst]: Here should be

    Then we connected :doc:`Linear Regression<../orange3/doc/visual-   programming/source/widgets/regression/linear>` learner to the **Polynomial Regression** widget.

File: doc/widgets/kmeans.rst (hunk 1 above)
Comment (ajdapretnar) [inline doc/widgets/kmeans.rst]: Agh, yes. There's a bit of a difference between Orange3 documentation style and add-ons doc style. It really depends on the markdown. It bugs me a bit that we have two different styles in our docs.... what's your opinion on this?

File: doc/widgets/kmeans.rst (hunk 2 above)
Comment (ajdapretnar) [inline doc/widgets/kmeans.rst]: My sloppiness, sorry. :see_no_evil:

File: doc/widgets/univariatepolynomialregression.rst (hunk 4 above)
Comment (ajdapretnar) [inline doc/widgets/univariatepolynomialregression.rst]: I was debating this myself. The main idea is to keep the signal part quite short, as explanation should come in the 'Description' section. Still not sure what is a better option, but after some discussion in our lab, I would prefer to keep it in the 'Description' part if you agree. :)

File: doc/widgets/univariatepolynomialregression.rst (hunk 5 above)
Comment (ajdapretnar) [inline doc/widgets/univariatepolynomialregression.rst]: Guh. Someone incredibly stupid is writing this. :disappointed: Will fix it.

**Comment by PrimozGodec** [comment]
Thanks @ajdapretnar

**Pull request merged**